device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -68,14 +69,8 @@
  * The task scheduler for sending and receiving messages for the Device Client
  */
 @Slf4j
-public final class DeviceIO
+public final class DeviceIO implements IotHubConnectionStatusChangeCallback
 {
-    /** The state of the IoT Hub client's connection with the IoT Hub. */
-    protected enum IotHubClientState
```

Comment:
This type was pretty redundant, and couldn't capture the same state as the IotHubTransport layer had, so I removed this type and just re-used the IotHubConnectionStatus class for this purpose